### ISSUE TITLE
Bus with lazy exporter?

### ISSUE THREAD

**Issue by nekobon** [InvestmentSystems/static-frame#256]
`Bus` is a great tool to load data lazily. It would be great if we can also create it lazily for exporting lazily.

For example, I have a generator that yields over 500 Frames. I'd like to save them in files, and eventually from another process, I would like to load some of them using a `Bus`. Currently in order to save them as a `Bus`, we can send the generator to the `Bus.from_frames` constructor but it would then keep all these 500 in memory.

Would it be possible to support lazy exporting? I'm not sure what the best way could be. Some random ideas are:
- `Bus.from_iterable()` that does not iterate until accessed or exported
- `Bus.from_frames_to_export(frame_gen, exporter=Bus.to_zip_tsv)` to construct and export at the same time

**Comment by flexatone** [comment]
Interesting suggestion! 

This idea of lazily working through an iterator is exactly what a `Batch` does. As `Bus` holds a `Series` of `Frame`, it needs to realize the entire iterator, in part because we need to discover the labels. While a `Bus` lazily loads `Frame` from disk, in all cases it needs to greedily have all labels available.

One option already available is to user the lower-level `Store` interfaces that back the `Bus`. For example:

```python
>>> from static_frame.core.store_zip import StoreZipPickle
>>> items = ((label, sf.Frame(np.arange(4).reshape(2, 2))) for label in ('a', 'b'))
>>> szp = StoreZipPickle('/tmp/tmp.zip')
>>> szp.write(items)
```

These StoreInterfaces, however, are not part of the public API. 

Maybe another option with an already public interface is to put compatible exporters on the `Batch`.

Let me think more about it, and see if others have input.

**Comment by flexatone** [comment]
I have extended the `Batch` interface to now expose the same `Store` based constructors and exporters  that are available on `Bus`. This means that a `Batch` can be used to lazily write into a `Store` given an iterator of labels, `Frame` pairs. The same `Store` can then be given to a `Bus` for lazy, random-access loading.

This will be available in the 0.6.35 release.

```python3 
>>> f1 = sf.Frame(np.arange(4).reshape(2, 2))
>>> f2 = sf.Frame(np.arange(4).reshape(2, 2)) * 10
>>> sf.Batch(zip(('a', 'b'), (f1, f2))).to_zip_parquet('/tmp/tmp.zip')
>>> sf.Bus.from_zip_parquet('/tmp/tmp.zip', config=sf.StoreConfig(index_depth=1))['b']
<Frame: b>
<Index>    0       1       <<U1>
<Index>
0          0       10
1          20      30
<int64>    <int64> <int64>
```
